daml/splice-wallet/daml/Splice/Wallet/MintingDelegation.daml

```diff
@@ -0,0 +1,132 @@
+-- Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Splice.Wallet.MintingDelegation where
+
+import DA.Assert
+import DA.Foldable (forA_)
+import Splice.AmuletRules
+import Splice.Round
+import Splice.Types
+import Splice.Util
+import Splice.Util.Token.Wallet.MergeDelegation (FeaturedAppRightCall, createActivityMarkerForProvider)
+import Splice.ValidatorLicense
+
+-- | Proposal by the user to setup a minting delegation.
+template MintingDelegationProposal
+  with
+    delegation : MintingDelegation
+      -- ^ The delegation to be created if accepted.
+  where
+    signatory delegation.beneficiary
+    observer delegation.delegate
+
+    choice MintingDelegationProposal_Accept : MintingDelegationProposal_AcceptResult
+      controller delegation.delegate
+      do MintingDelegationProposal_AcceptResult <$> create delegation
+
+    choice MintingDelegationProposal_Reject : MintingDelegationProposal_RejectResult
+      controller delegation.delegate
+      do pure MintingDelegationProposal_RejectResult with dummyUnitField = ()
+
+    choice MintingDelegationProposal_Withdraw : MintingDelegationProposal_WithdrawResult
+      controller delegation.beneficiary
+      do pure MintingDelegationProposal_WithdrawResult with dummyUnitField = ()
+
+defaultAmuletMergeLimit : Int
+defaultAmuletMergeLimit = 10
+
+-- | The right allowing delegate to record liveness activity and mint rewards on behalf of the beneficiary.
+template MintingDelegation
+  with
+    beneficiary : Party
+      -- ^ The party on whose behalf minting is performed.
+    delegate : Party
+      -- ^ The party authorized to perform minting operations.
+    dso : Party
+      -- ^ Expected DSO party.
+    expiresAt : Time
+      -- ^ The time after which this delegation is no longer valid.
+    amuletMergeLimit : Int
+      -- ^ The number of amulet contracts to keep after auto-merging; i.e.,
+      -- auto-merge contracts once there are strictly more than this number of contracts.
+  where
+    signatory beneficiary, delegate
+
+    nonconsuming choice MintingDelegation_RecordValidatorLivenessActivity : MintingDelegation_RecordValidatorLivenessActivityResult
+      with
+        validatorLicenseCid : ContractId ValidatorLicense
+          -- ^ The validator license of the beneficiary to record liveness activity for.
+        openRoundCid : ContractId OpenMiningRound
+          -- ^ The open mining round for which to record activity.
+        optFeaturedAppRight : Optional FeaturedAppRightCall
+          -- ^ The featured app right to use to feature the delegate's activity.
```

Comment:
Upon further consideration, I think this goes in the wrong direction. The direction of "Minting rewards for minting rewards" seems problematic.

More generally, the contracts here implement purely operational tools, and thus do not qualify for app rewards. I'd thus suggest to drop the app reward support here.

This also makes it consistent with the choice below.

Reply:
@meiersi-da I have removed the `FeaturedAppRight` related code. The PR is ready for review